src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,9 +88,20 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
-        try (final Writer fileWriter =
-                     new BufferedWriter(new FileWriter(path.toFile()))) {
+        try {
+            if (!path.toFile().exists()) {
+                throw new FileNotFoundException();
+            }
+            writeFile(addressBook);
+        } catch (FileNotFoundException fnfe) {
+            System.out.println("Storage file deleted! Recreating...");
```

Comment:
what happens to SLAP when you print here? think!

there is already a class that performs user interaction, think how and where you can use that to show the error message?

i like this approach :)